src/main/kotlin/me/kpavlov/ksp/maven/AbstractKspProcessMojo.kt

```
@@ -272,16 +307,25 @@ class KspProcessMojo(
-            val processing =
+        // Create a new KotlinSymbolProcessing instance for each execution
+        // to ensure complete isolation in parallel builds
+        val processing =
+            try {
                 kspFactory.create(
                     kspConfig = kspConfig,
                     symbolProcessorProviders = processorProviders,
-                    logger = kspLogger,
+                    logger = KspLogger(log = log, scope = scope),
                 )
+            } catch (ex: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)